presentation/src/main/java/me/androidbox/presentation/navigation/NavigationGraph.kt

```diff
@@ -0,0 +1,97 @@
+import me.androidbox.presentation.login.screen.RegisterScreen
+import me.androidbox.presentation.login.viewmodel.LoginViewModel
+import me.androidbox.presentation.login.viewmodel.RegisterViewModel
+
+@Composable
+fun NavigationGraph(
+    navHostController: NavHostController
+) {
+    NavHost(
+        navController = navHostController,
+        startDestination = Screen.LoginScreen.route
+    ) {
+
+        /* LoginScreen */
+        composable(
+            route = Screen.LoginScreen.route
+        ) {
+            val loginViewModel: LoginViewModel = hiltViewModel()
+            val loginState = loginViewModel.loginState.collectAsState()
+
+            LoginScreen(
+                loginState = loginState,
+                onSignUpClicked = {
+                    /* Signup clicked, navigate to register screen */
+                    navHostController.navigate(route = Screen.RegisterScreen.route)
+                },
+                onLoginSuccess = { login ->
+                    loginViewModel.saveCurrentUserDetails(login)
+                    /* TODO Navigate to the agenda screen (not implemented yet) */
+                },
+                email = loginViewModel.email,
+                onEmailChanged = { newEmail ->
+                    loginViewModel.onEmailChanged(newEmail)
+                },
+                password = loginViewModel.password,
+                onPasswordChanged = { newPassword ->
+                    loginViewModel.onPasswordChanged(newPassword)
+                },
+                onPasswordVisibilityChanged = {
+                    loginViewModel.onPasswordVisibilityChanged()
+                },
+                isPasswordVisible = loginViewModel.isPasswordVisible,
+                onLoginUser = { email, password ->
+                    loginViewModel.loginUser(email, password)
+                }
+            )
+        }
+
+        /* Register Screen */
+        composable(
+            route = Screen.RegisterScreen.route
+        ) {
+            val registerViewModel: RegisterViewModel = hiltViewModel()
+            val registerState = registerViewModel.registrationState.collectAsState()
+
+            RegisterScreen(
+                registrationState = registerState,
+                onBackArrowClicked = {
+                    /* Back arrow clicked, pop RegisterScreen of the backstack to get back to login screen */
+                    navHostController.popBackStack()
+                },
+                onRegistrationSuccess = {
+                    /* Registration Success */
+                    navHostController.popBackStack()
+                },
+                username = registerViewModel.username.collectAsState().value,
+                email = registerViewModel.email.collectAsState().value,
+                password = registerViewModel.password.collectAsState().value,
+                isPasswordVisible = registerViewModel.isPasswordVisible.collectAsState().value,
```

Comment:
Is this ok to do like this `registerViewModel.isPasswordVisible.collectAsState().value`

Reply:
I personally like to have just 1 state, and then act based on it

Ej: RegisterState

Reply:
Yeah, as we talked about in today's call having one state class synergizes well with Compose projects. Not mandatory, but I don't see a reason why you would not want to do that.
Also, IMO this reads better:
```
val state by viewModels.state.collectAsState()
YourScreen(
    state = state
)
```

compared to `state = viewModel.state.collectAsState().value`